### PR TITLE
Support for notification orders requiring organization lookup

### PR DESCRIPTION
**Pull request by acn-sbuad**
## Description

- Organisation -> Organization
- Possible to post order with organization number
- Implemented lookup of contact points for org on order placement and notification generation
- Logic for validating and manipulating mobile numbers moved to core 

## Related Issue(s)
- #456 

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [x] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/Altinn.Notifications.Core/Services/ContactPointService.cs [hunk 1]
```diff
@@ -74,51 +90,73 @@ private async Task<List<Recipient>> AugmentRecipients(List<Recipient> recipients
...
+            /* the output from this function should include an AUHTORIZED list of user registered contact points if notification has a service affiliation 
+                will require the extension of the OrganizationContactPoints class */
+            List<string> orgNos = recipients
+             .Where(r => !string.IsNullOrEmpty(r.OrganizationNumber))
+             .Select(r => r.OrganizationNumber!)
+             .ToList();
+
+            if (orgNos.Count == 0)
+            {
+                return [];
```
Comment (SandGrainOne) [inline src/Altinn.Notifications.Core/Services/ContactPointService.cs:144]: Should we use explicit type here? It's explicit a bit further up.

Comment (acn-sbuad) [reply src/Altinn.Notifications.Core/Services/ContactPointService.cs:144]: rather changed the other function to return [] as well. Shorter notation and I think return type is implicit.

File: test/Altinn.Notifications.Tests/Notifications.Core/TestingHelpers/MobileNumberHelperTests.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    [InlineData("90000000", false)]
+    [InlineData("+4790000000", true)]
+    [InlineData("+4750000004", false)]
+    [InlineData("+47900000001", false)]
+    [InlineData("+14790000000", false)]
+    [InlineData("004790000002", true)]
+    [InlineData("", false)]
+    [InlineData("111100000", false)]
+    [InlineData("dasdsadSASA", false)]
+    public void IsValidMobileNumber(string mobileNumber, bool expectedResult)
```
[the remark this answers is not in the document]
Comment (acn-sbuad) [reply test/Altinn.Notifications.Tests/Notifications.Core/TestingHelpers/MobileNumberHelperTests.cs:38]: added test case

File: src/Altinn.Notifications.Integrations/Register/OrgContactPointLookup.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
+﻿using System.Text.Json.Serialization;
+
+namespace Altinn.Notifications.Integrations.Register
+{
+    /// <summary>
+    /// A class describing the query model for contact points for organizations
+    /// </summary>
+    public class OrgContactPointLookup
```
[the remark this answers is not in the document]
Comment (acn-sbuad) [reply src/Altinn.Notifications.Integrations/Register/OrgContactPointLookup.cs:8]: image
Having a hard time concluding anything on this. As it stands now it is consistent with other lookups, but we use both contactPoint and contactPoints..

**Review by SandGrainOne — APPROVED** [review]

**Pull request merged**